### DISCUSSION TITLE
GNU's libstdc++ or clang libc++

### DISCUSSION THREAD

**Discussion by viniciusferrao** [zauguin/i18n-cpp#9] [Q&A] [answered]
Hi @zauguin which standard library do you used to develop this library?

I have an error trying to compile it under macOS (again) but is seems to be an libc++ issue.

```
FAILED: merge/CMakeFiles/i18n-merge-pot.dir/merge.cpp.o 
/opt/local/bin/clang++ -DFMT_LOCALE -DFMT_SHARED -isystem /opt/local/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -std=c++20 -MD -MT merge/CMakeFiles/i18n-merge-pot.dir/merge.cpp.o -MF merge/CMakeFiles/i18n-merge-pot.dir/merge.cpp.o.d -o merge/CMakeFiles/i18n-merge-pot.dir/merge.cpp.o -c /Users/ferrao/Documents/Development/i18n++/merge/merge.cpp
/Users/ferrao/Documents/Development/i18n++/merge/merge.cpp:15:44: error: invalid operands to binary expression ('const std::optional<std::string>' (aka 'const optional<basic_string<char, char_traits<char>, allocator<char>>>') and 'const std::optional<std::string>')
    std::strong_ordering order = a.context <=> b.context;
                                 ~~~~~~~~~ ^   ~~~~~~~~~
```

The spaceship operator `<=>` is available on Clang but not on `libc++`.

From the `<compare>` library you see that `LLVM` does not implement the feature [__cpp_lib_three_way_comparison](https://github.com/llvm/llvm-project/blob/main/libcxx/include/version#L368)

On `libstdc++` it's properly defined [here](https://github.com/gcc-mirror/gcc/blob/16e2427f50c208dfe07d07f18009969502c25dc8/libstdc%2B%2B-v3/include/std/version#L192).

**Comment by zauguin** [comment, marked as answer]
As you seem to have guessed I'm mostly using libstdc++ and while I try to avoid any libstdc++ specific dependencies, I tend to assume that all standard libraries are *at least as C++20 compliant as* libstdc++. For libc++ this assumption is (sadly very) wrong.

Anyway, it should be fixed. Can you add an issue for this?

**Reply by viniciusferrao** [reply]
Opened #10
